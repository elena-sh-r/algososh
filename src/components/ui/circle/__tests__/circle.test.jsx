import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { ElementStates } from '../../../../types/element-states';
import { Circle } from '../circle';

describe('Тестирование компонента Circle', () => {
  it('Элемент без буквы отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  }); 

  it('Элемент с буквами отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle letter="АБВ" />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });
  
  it('Элемент с head отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle head="АБВ" />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Элемент с react-элементом в head отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle head={<div>React-элемент</div>}/>)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Элемент с tail отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle tail="АБВ" />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Элемент с react-элементом в tail отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle tail={<div>React-элемент</div>}/>)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Элемент с index отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle index={123} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Элемент с пропом isSmall ===  true отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle isSmall={true} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Элемент в состоянии default отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Элемент в состоянии changing отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });

  it('Элемент в состоянии modified отрисовывается корректно', () => {
    const circle = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
      expect(circle).toMatchSnapshot();
  });
});