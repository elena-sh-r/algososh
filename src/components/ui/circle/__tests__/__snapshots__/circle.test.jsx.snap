// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle Элемент без буквы отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Элемент в состоянии changing отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Элемент в состоянии default отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Элемент в состоянии modified отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Элемент с head отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    АБВ
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Элемент с index отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    123
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Элемент с react-элементом в head отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      React-элемент
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Элемент с react-элементом в tail отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      React-элемент
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle Элемент с tail отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    АБВ
  </div>
</div>
`;

exports[`Тестирование компонента Circle Элемент с буквами отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      АБВ
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle Элемент с пропом isSmall ===  true отрисовывается корректно 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
